core/src/plugins/filed/BareosFdPluginPostgres.py

```diff
@@ -0,0 +1,444 @@
+            self.dbuser = self.options["dbuser"]
+        else:
+            self.dbuser = "root"
+        if not "switchWal" in self.options:
+            self.switchWal = True
+        else:
+            self.switchWal = self.options["switchWal"].lower() == "true"
+        return bRCs["bRC_OK"]
+
+    def start_backup_job(self, context):
+        """
+        Make filelist in super class and tell Postgres
+        that we start a backup now
+        """
+        bareosfd.DebugMessage(
+            context, 100, "start_backup_job in PostgresPlugin called",
+        )
+        try:
+            self.dbCon = psycopg2.connect(
+                "dbname=%s user=%s" % (self.dbname, self.dbuser)
+            )
+            self.dbCursor = self.dbCon.cursor()
+            self.dbCursor.execute("SELECT current_setting('server_version_num')")
+            self.pgVersion = int(self.dbCursor.fetchone()[0])
+            #bareosfd.DebugMessage(
+            #    context, 1, "Connected to Postgres version %d\n" % self.pgVersion,
+            #)
+            ## WARNING: JobMessages cause fatal errors at this stage
+            JobMessage(
+               context,
+               bJobMessageType["M_INFO"],
+               "Connected to Postgres version %d\n"
+               % (self.pgVersion),
+            )
+        except:
+            bareosfd.JobMessage(
+                context,
+                bJobMessageType["M_FATAL"],
+                "Could not connect to database %s, user %s\n"
+                % (self.dbname, self.dbuser),
+            )
+            return bRCs["bRC_Error"]
+        # If level is not Full, we only backup newer WAL files
+        if chr(self.level) == "F":
+            startDir = self.options["postgresDataDir"]
+            self.files_to_backup.append(startDir)
+            bareosfd.DebugMessage(
+                context, 100, "dataDir: %s\n" % self.options["postgresDataDir"],
+            )
+        else:
+            # For Full the Restore object comes after DB connection close
+            self.files_to_backup.append("ROP")
+            # With incremental / diff jobs we just backup new WAL files
+            startDir = self.options["walArchive"]
+            # get current Log Sequence Number (LSN)
+            # PG8: not supported
+            # PG9: pg_get_current_xlog_location
+            # PG10: pg_current_wal_lsn
+            pgMajorVersion = self.pgVersion // 10000
+            if pgMajorVersion >= 10:
+                getLsnStmt = "SELECT pg_current_wal_lsn()"
+                switchLsnStmt = "SELECT pg_switch_wal()"
+            elif pgMajorVersion >= 9:
+                getLsnStmt = "SELECT pg_current_xlog_location()"
+                switchLsnStmt = "SELECT pg_switch_xlog()"
+            if pgMajorVersion < 9:
+                bareosfd.JobMessage(
+                    context,
+                    bJobMessageType["M_INFO"],
+                    "WAL switching not supported on Postgres Version < 9\n",
+                )
```

Comment:
No return value here? Shouldn't it return FAILED or something?

Reply:
Backup itself does still work. It's only the forced wal-switch before an incremental backup is not supported before version 9. It is just this option not working.